packages/flutter/lib/src/widgets/parent_child_checkbox.dart

```diff
@@ -0,0 +1,183 @@
+import 'package:flutter/material.dart';
+
+///Main widget of Parent Child Checkbox.
+///Parent Child Checkbox is a type of checkbox where we can establish hierarchy in Checkboxes.
+class ParentChildCheckbox extends StatefulWidget {
+  ///Text Widget to specify the Parent checkbox
+  final Text? parent;
+
+  ///List<Text> Widgets to specify the Children checkboxes
+  final List<Text>? children;
```

Comment:
Forcing these to be Text widgets is against Flutter style: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-interleaving-multiple-concepts-together